src/metabase/integrations/slack.clj

```diff
@@ -118,15 +118,20 @@
 
 (defn- handle-error [body]
   (let [invalid-token? (slack-token-error-codes (:error body))
-        message        (if invalid-token?
-                         "Invalid token"
-                         (format "Slack API error: %s" (:error body)))
-        error          (if invalid-token?
-                         {:error-code (:error body)
-                          :errors     {:slack-token message}}
-                         {:error-code (:error body)
-                          :message    message
-                          :response   body})]
+        missing-channel? (= (:error body) "channel_not_found")
+        message        (cond
+                         invalid-token? "Invalid token"
+                         missing-channel? "Channel not found"
+                         :else (format "Slack API error: %s" (:error body)))
+        error          (cond
+                         invalid-token? {:error-code (:error body)
+                                         :errors     {:slack-token message}}
```

Comment:
I know it's an existing pattern but I don't think having the message here is useful. I don't think we use it anywhere.

In fact, inside `metabase.notification.send/should-retry-sending?` we're using the existent of a key to determine the error type.

I propose we instead replace this errors map with an `:error-type` key with 3 possible values: `:slack/invalid-token, :slack/channel-not-found, :slack/unknown`.

Let me know if you're interested in improving this.